distributed/worker.py

```diff
@@ -739,7 +747,9 @@ def __init__(
             )
 
         self.batched_stream = BatchedSend(interval="2ms", loop=self.loop)
-        self.name = name
+        self.name = dask.config.get(
+            "distributed.worker.name", override_with=name
+        ).format(**os.environ)
```

Comment:
In some cases it would be valuable to pass a format string here. Using the container hostname as the worker name is one example. So I've added an extra env var format expansion here.

Reply:
Should we move this up into `dask.config.get` ?

We would check for `isinstance(value, str) and "$" in value` and if so expand with `.format` as you've done here.

Reply:
Sure that sounds like a nice enhancement. I'll leave that for a follow up PR though rather than increasing scope here.